script/testing/util/db_server.py

```diff
@@ -227,21 +228,220 @@ def get_build_path(build_type):
+# Value Preprocessors
+#
+# The signature for value preprocessors must be:
+#   def preprocessor(value: str, meta: Dict) -> str:
+#       ...
+
+@applies_to(bool)
+def lower_booleans(value, meta):
+    """
+    Lower boolean string values to the format expected by the DBMS server.
+    
+    e.g. 
+        `True` -> `true`
+        `False` -> `false`
+    
+    Arguments
+    ---------
+    value : str 
+        The DBMS server argument value
+    meta : Dict
+        Dictionary of meta-information available to all preprocessors
+
+    Returns
+    -------
+    The preprocessed server argument value
+    """
+    return str(value).lower()
 
-    return ' '.join(server_args_arr)
+@applies_to(str)
+def resolve_relative_paths(value, meta):
+    """
+    Resolve relative paths in the DBMS server arguments to their equivalent absolute paths.
+
+    When specifying path arguments to the DBMS, it is often simpler to think in terms
+    of how the path relates to the location of the DBMS server binary. However, because
+    the command used by NoisePageServer to launch the DBMS server instance specifies
+    the absolute path to the binary, relative paths (that necessarily encode the expected
+    location of the binary) will fail to function properly. Expanding these relative paths
+    to their absolute counterparts before passing them to the DBMS addresses this issue.
+
+    The above highlights the fundamental limitation of this current implementation, namely:
+
+        ALL RELATIVE PATHS ARE ASSUMED TO BE RELATIVE TO THE DBMS BINARY DIRECTORY
+
+    All relative paths that are relative to another directory will fail to resolve properly.
+
+    Arguments
+    ---------
+    value : str 
+        The DBMS server argument value
+    meta : Dict
+        Dictionary of meta-information available to all preprocessors
+
+    Returns
+    -------
+    The preprocessed server argument value
+    """
+    is_relative = str.startswith(value, "./") or str.startswith(value, "../")
+    # TODO(Kyle): This doesn't actually do any "resolving", it merely appends
+    # the absolute path to the binary directory to the relative path as it 
+    # presently exists; this works fine, but also means we might pass a needlessly
+    # long string to the DBMS server e.g. bin/././././whatever.txt is possible.
+    # It might be worth it to find some portable way to completely resolve these.
+    return os.path.join(meta["bin_dir"], value) if is_relative else value
+
+@applies_to(int, str, bool)
+def handle_flags(value, meta):
+    """
+    Handle DBMS server arguments with no associated value.
+
+    Some arguments to the DBMS are flags that do not have an associated value;
+    in these cases, we do not want to format the complete argument as
+    `-attribute=value` and instead want to format it as `-attribute` alone.
+    This preprocessor encapsulates the logic for this transformation.
+
+    TODO(Kyle): Do we actually support any arguments like this? 
```

Comment:
See #1388 and make sure it somewhat matches behavior.

Reply:
Incidentally, is this the expected behavior?
```
>>> construct_server_argument('wal_disabled', False, {})
'-wal_disabled=false'
>>> construct_server_argument('wal_disabled', True, {})
'-wal_disabled=true'
>>> construct_server_argument('wal_disabled', None, {})
'-wal_disabledNone'
```

Reply:
No, that is a necessary case that I didn't consider. Have to do a little hacking around because we aren't allowed to manually specify `NoneType` (the type of the singleton `None`), but I updated to work around the issue such that we now get:

```
>>> construct_server_argument('wal_disabled', False, {})
'-wal_disabled=false'
>>> construct_server_argument('wal_disabled', True, {})
'-wal_disabled=true'
>>> construct_server_argument('wal_disabled', None, {})
'-wal_disabled'
```